narwhal/network/src/connectivity.rs

```diff
@@ -19,12 +28,14 @@ impl ConnectionMonitor {
         network: anemo::NetworkRef,
         connection_metrics: NetworkConnectionMetrics,
         peer_id_types: HashMap<PeerId, String>,
+        sender: Option<Sender<(PeerId, ConnectionStatus)>>,
```

Comment:
Having an `mpsc sender` here seems to me quite restrictive an tailoring the solution for our case. I would suggest if you want to stick to that implementation, then changing to a [broadcast](https://docs.rs/tokio/latest/tokio/sync/broadcast/) instead might allow more to consume the connectivity status and let the `ConnectionMonitor` play a wider role in the codebase